hazelcast/src/main/java/com/hazelcast/config/security/SimpleAuthenticationConfig.java

```diff
@@ -0,0 +1,274 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config.security;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.hazelcast.config.LoginModuleConfig;
+import com.hazelcast.config.LoginModuleConfig.LoginModuleUsage;
+
+/**
+ * Typed authentication configuration for {@code SimplePropertiesLoginModule}. The user configuration (username, password, and
+ * assigned roles) is directly part of the configuration object.
+ */
+public class SimpleAuthenticationConfig extends AbstractClusterLoginConfig<SimpleAuthenticationConfig> {
+
+    private final Map<String, UserDto> userMap = new ConcurrentHashMap<>();
+    private volatile String roleSeparator;
+
+    /**
+     * Adds one user to the configuration.
+     *
+     * @param username username
+     * @param password user's password
+     * @param roles roles assigned to the user
+     * @return this object
+     */
+    public SimpleAuthenticationConfig addUser(@Nonnull String username, @Nonnull String password, String... roles) {
+        userMap.computeIfAbsent(requireNonNull(username), u -> new UserDto(password)).setRoles(roles);
+        return self();
+    }
+
+    /**
+     * Adds one user to the configuration.
+     *
+     * @param username username
+     * @param userDto user's attributes
+     * @return this object
+     */
+    public SimpleAuthenticationConfig addUser(@Nonnull String username, @Nonnull UserDto userDto) {
+        userMap.put(requireNonNull(username, "Username has to be provided"),
+                requireNonNull(userDto, "UserDto object has to be provided"));
+        return self();
+    }
+
+    /**
+     * Allows to use a custom role separator in the configuration. It's important in case when a role name contains the default
+     * separator "," (comma).
+     *
+     * @param roleSeparator new separator
+     * @return this object
+     */
+    public SimpleAuthenticationConfig setRoleSeparator(@Nullable String roleSeparator) {
+        this.roleSeparator = roleSeparator;
+        return self();
+    }
+
+    /**
+     * Returns the custom role separator (when set). If no custom one is used, {@code null} is returned and default separator is
+     * used ("," - comma).
+     *
+     * @return the separator
+     */
+    public String getRoleSeparator() {
```

Comment:
nit: annotate as `@Nullable`? or maybe it can just return `","` when no separator is defined by user?

Reply:
:+1: I'll add the `@Nullable`.